lib/sycamore/sycamore/plan_nodes.py

```diff
@@ -3,20 +3,37 @@
 
 if TYPE_CHECKING:
     from ray import Dataset
+    from sycamore.context import Context
 
 
 class NodeTraverse:
     def __init__(
-        self, before: Optional[Callable[["Node"], "Node"]] = None, after: Optional[Callable[["Node"], "Node"]] = None
+        self,
+        before: Optional[Callable[["Node"], "Node"]] = None,
+        visit: Optional[Callable[["Node"], None]] = None,
```

Comment:
why does this not return the node either?

Reply:
I don't think I understand the poin tof visit? Can you add a couple comments?

Reply:
Simpler code.